generate_outputs_t5.py

```diff
@@ -0,0 +1,40 @@
+from datasets import ClassLabel, load_dataset, Value 
+from transformers import AutoTokenizer, T5ForConditionalGeneration
+import torch
+import argparse
+
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+
+if __name__=='__main__':
+    parser = argparse.ArgumentParser(description='Specify output path.')
+    parser.add_argument("--output_dir", type=str, required=True, help="Directory of the model.")
+    parser.add_argument("--checkpoint", type=str, required=True, help="Model checkpoint.")
+    args = parser.parse_args()
+
+    #load model
+    tokenizer = AutoTokenizer.from_pretrained(args.output_dir)
+    tokenizer.pad_token = tokenizer.eos_token
+    model = T5ForConditionalGeneration.from_pretrained(args.output_dir+'/'+args.checkpoint).to(device)
+
+    #load input data
+    f = open('test_mrs.txt', 'r+')
+    ls = f.readlines()
+    f.close()
+    outputs= []
+    for i in range(len(ls)):
+        ls[i] = ls[i].replace('\n','')
+        txt = ls[i] + " <ref:> "
+        inputs = tokenizer(txt, return_tensors='pt').to(device)
+
+        o=model.generate(
+            **inputs, 
+            num_beams=10, 
+            length_penalty=2, 
+            max_length=512)
+        output_text = tokenizer.decode(o[0], skip_special_tokens=True)
```

Comment:
Make this a separate function
`generate(model, tokenizer, input: str) -> str:`

That way it is easier to see what is the input and what is the output.

Moreover, why don't you iterate directly over `ls`?

```
outputs = [generate(model, tokenizer, input) + '\n' for input in ls] 
```